chopper_generator/lib/src/generator.dart

```diff
@@ -550,7 +550,12 @@ final class ChopperGenerator
       }
     }
 
-    return _generateUri('$finalBaseUrl$path'.replaceAll('//', '/'));
+    return _generateUri(
+      '$finalBaseUrl$path'.replaceAll(
+        RegExp(r'(?<!http:|https:)\/\/'),
```

Comment:
I just put this REGEX into regexr.com and have seen that it will match:

- https:`//`test.com/foo/bar/baz
- http://test.com/foo/bar/baz
- https:`//`test.com`//`foo`//`bar`//`baz
- http://test.com`//`foo`//`bar`//`baz

Is this expected?

Reply:
look i did the same and works as expected

![image](https://github.com/lejard-h/chopper/assets/56127727/74d1dbd3-6a90-4eeb-b4b7-283bbcb2cefa)

maybe is your browser, could you please tell me what browser are you using?

Reply:
The latest Safari.

Reply:
I still find this regex too cryptic for my taste. Have you tried this approach

1. extract the protocol using `Uri.scheme`
2. find how long the scheme and intial `//` is, i.e. 6 or 7 chars
3. use `substring` to replace all `//` with `/` in the remainder od the string without the protocol and initial `//`

Reply:
ok bro i will implement that solution

Reply:
ready